deepchem/models/losses.py

```diff
@@ -187,7 +187,12 @@ class SparseSoftmaxCrossEntropy(Loss):
 
   def _compute_tf_loss(self, output, labels):
     import tensorflow as tf
+
+    if labels.shape[-1] == 1:
+      labels = tf.squeeze(labels, axis=-1)
```

Comment:
I don't think that's the right check for TensorFlow.  `sparse_softmax_cross_entropy_with_logits()` requires `labels` to have one fewer dimension than `output`.  We should check whether `len(labels.shape) == len(output.shape)+1`.

Reply:
How about adding an additional assertion statement after L192 for the check you proposed? There can be multiple reasons for this. L191/192 are mainly handling the issue in molnet data loader.

Reply:
It's not an assertion.  The issue is that this function is more permissive of its inputs than the TensorFlow and PyTorch functions we use to implement it.  Given any inputs that are legal for this function, we need to reshape them as required for the TensorFlow and PyTorch functions.  Conversely, we should only reshape them if the inputs are *not* the correct shape for the implementing function.  If the inputs already have the shape expected by `sparse_softmax_cross_entropy_with_logits()`, we shouldn't call `squeeze()`.

Reply:
What check/reshape will you suggest? It's not guaranteed that one can obtain `len(labels.shape) == len(output.shape)+1` by reshaping as the inputs may not be valid themselves.

Reply:
If the inputs aren't valid, that's the user's problem, not ours. 😀  How about

```python
if len(labels.shape) == len(outputs.shape):
  labels = tf.squeeze(labels, axis=-1) 
```

Reply:
Sounds good. I've made the update.

Reply:
With "thumbs up", I assume it looks good to you 😄 . Any further issues? @peastman

Reply:
I think it's good to merge.